docs/extenders/Error Handling.md

```diff
@@ -0,0 +1,38 @@
+# Error Handling Best Practices v1
+
+This document describes error handling best practices for Zowe Explorer extenders.
+
+## Contents
+
+- [Objectives](#objectives)
+- [Error Message Format](#error-message-format)
+- [Special cases](#special-cases)
+
+## Objectives
+
+Customers and developers need to see clear error messages when they use Zowe Explorer and its extenders.
+
+By following this documentation, Zowe Explorer extenders will help users debug their extensions quickly and easily.
+
+Zowe Explorer encourages extenders to use this guidance in order to:
+- Help users understand which extension is causing an error, by including the name of the extension in the error message.
+- Give developers more information in error logs.
```

Comment:
Why we said, that we have 3 objectives, but mention only one of them below?

Just a suggestion: I would not recommend to include **logging** objective in the document about **error handling**. 

And the third objective is not covered at all... 
Just a suggestion: Maybe we can provide some practices or special keywords to include into the error messages to make them more understandable? Maybe patterns or smt like this...

Reply:
By i18n, do you mean localization?

Reply:
Yep

Reply:
Also I think eventually we should have logging & error handling in the same document. Maybe named appropriately. Otherwise both documents will be really short. If I was extending Zowe Explorer I would be annoyed to have to open a bunch of different files in different windows just to read closely-related information.

Reply:
And, the third objective is covered by including the extension name in the message, then "Error", then the error itself. This is the format that will be standard.